llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2772,8 +2784,11 @@ static bool isKnownNonZeroFromOperator(const Operator *I,
     //    This all implies the 2 i16 elements are non-zero.
     Type *FromTy = I->getOperand(0)->getType();
     if ((FromTy->isIntOrIntVectorTy() || FromTy->isPtrOrPtrVectorTy()) &&
-        (BitWidth % getBitWidth(FromTy->getScalarType(), Q.DL)) == 0)
+        (BitWidth % getBitWidth(FromTy->getScalarType(), Q.DL)) == 0) {
+      if (match(I, m_ElementWiseBitCast(m_Value())))
+        return isKnownNonZero(I->getOperand(0), DemandedElts, Q, Depth);
```

Comment:
I don't think this case is possible? You'd need an element-wise bitcast involving only pointer or integer types. This means it's a pointer-to-pointer bitcast or an integer-to-integer bitcast, which means it must be a trivial bitcast, that we shouldn't bother handling specially.